pallet-logion-loc/src/lib.rs

```diff
@@ -1707,23 +1741,49 @@ pub mod pallet {
                 && collection_loc.void_info.is_none()
         }
 
-        fn can_submit(loc_id: &T::LocId, loc: &LegalOfficerCaseOf<T>, submitter: &SupportedAccountId<T::AccountId, T::EthereumAddress>) -> bool {
-            match &submitter {
-                SupportedAccountId::Polkadot(pokadot_submitter) => *pokadot_submitter == loc.owner
-                    || match &loc.requester {
-                        Account(requester_account) => *pokadot_submitter == *requester_account,
-                        _ => false
-                    }
-                    || Self::verified_issuers_by_loc(loc_id, pokadot_submitter).is_some(),
-                SupportedAccountId::Other(other_submitter) => match &other_submitter {
-                    OtherAccountId::Ethereum(ethereum_submitter) => match &loc.requester {
-                        Requester::OtherAccount(other_requester) => match &other_requester {
-                            OtherAccountId::Ethereum(ethereum_requester) => *ethereum_submitter == *ethereum_requester,
+        fn is_valid_submitter(loc_id: &T::LocId, loc: &LegalOfficerCaseOf<T>, submitter: &SupportedAccountId<T::AccountId, T::EthereumAddress>, published_by_owner: bool) -> bool {
+
+            let polkadot_requester: Option<&T::AccountId> = match &loc.requester {
+                Account(requester_account) => Some(requester_account),
+                _ => None
+            };
+            if published_by_owner {
+                match &submitter {
+                    SupportedAccountId::Polkadot(polkadot_submitter) => *polkadot_submitter == loc.owner ||
+                        (polkadot_requester.is_none() && Self::verified_issuers_by_loc(loc_id, polkadot_submitter).is_some()),
```

Comment:
As mentioned in the comment on the PR, I would remove the second term of the OR (i.e. only allow publication by owner with non-Polkadot requesters).